### PR TITLE
POL-35 AWS Publicly Accessible RDS Instances

### PR DESCRIPTION
**Pull request by cvurubindi**
Initial release for POL-35 AWS Publicly Accessible RDS Instances

### Description

This policy checks all Relational Database Service (RDS) instances and reports on any that are publicly accessible. When such an instance is detected, the user can choose to disable the publicly accessible option and the user can also choose to delete by enabling 'delete action' option as mentioned in Enable delete action section below.

### Issues Resolved

When a publicly accessible RDS instance is detected, an email action is triggered automatically to notify the specified users of the incident. Users then have an option to modify configuration after manual approval, and even Users can perform delete action if required. 
- *remove public access rule* - modifies the configuration of the RDS instance by disabling the public access rule

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] New functionality has been documented in CHANGELOG.MD

### Running Template Link:
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5cdd360938e25e007152d7c0

### Screenshot:
image

### CODE REVIEW

File: security/aws/rds_publicly_accessible/AWS_Publicly_Accessible_RDS_Instances.pt [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+  type "list"
+  label "Tags to ignore"
+  description "List of tags that will exclude resources from being evaluated by this policy. Multiple tags are evaluated as an 'OR' condition. Tag keys or key/value pairs can be listed. Example: 'test,env=dev'"
+end
+
+###############################################################################
+# Authentication
+###############################################################################
+
+auth "auth_us_east_1", type: "aws" do
```
Comment (avinashbhashyam-rs) [inline security/aws/rds_publicly_accessible/AWS_Publicly_Accessible_RDS_Instances.pt]: take a look at https://github.com/rightscale/policy_templates/pull/234#discussion_r284825518. You can just use one auth block.

File: security/aws/rds_publicly_accessible/AWS_Publicly_Accessible_RDS_Instances.pt [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+#To get list of All RDS Instances.
+datasource "ds_rds_instances_list" do
+  iterate $ds_regions_list
+  request do
+    run_script $js_rds_list, val(iter_item,"region")
+  end
+  result do
+   encoding "json"
+    collect jmes_path(response, "DescribeDBInstancesResponse.DescribeDBInstancesResult.DBInstances[*]") do
+      field "dbInstanceArn", jmes_path(col_item, "DBInstanceArn")
```
Comment (avinashbhashyam-rs) [inline security/aws/rds_publicly_accessible/AWS_Publicly_Accessible_RDS_Instances.pt]: be consistent with casing. I would suggest stick to `snake_case`

File: security/aws/rds_publicly_accessible/AWS_Publicly_Accessible_RDS_Instances.pt (hunk 1 above)
Comment (cvurubindi) [reply security/aws/rds_publicly_accessible/AWS_Publicly_Accessible_RDS_Instances.pt]: Looked at #234 (comment) and taken care of them. 
Thanks for suggestion, used one auth block and tested it, working fine

File: security/aws/rds_publicly_accessible/AWS_Publicly_Accessible_RDS_Instances.pt (hunk 2 above)
Comment (cvurubindi) [reply security/aws/rds_publicly_accessible/AWS_Publicly_Accessible_RDS_Instances.pt]: Thanks, I corrected that now.

**Comment by cvurubindi** [comment]
@rshade - Sorry for the inconvenience caused. 
I re-submitted the changes for valid check condition as per the comments given in the #234 (For check val), Pls review. Thanks.

**Pull request merged**